### DISCUSSION TITLE
How to render smoke in mitsuba 3

### DISCUSSION THREAD

**Discussion by LawrenceShum** [mitsuba-renderer/mitsuba3#624] [Q&A] [answered]
I have gridvolume file .vol to represent density of smoke.
The parameter "density" in heterogeneous medium exists in Mitsuba 0.5.0 has been removed in Mitsuba 3.
How to smoke in Mitsuba 3? 
Thanks!

**Comment by njroussel** [comment, marked as answer]
The `heterogenous` medium is documented [here](https://mitsuba.readthedocs.io/en/stable/src/generated/plugins_media.html#heterogeneous-medium-heterogeneous).
If you're unfamiliar with the terms please have a look at [this section of PBRT](https://pbr-book.org/3ed-2018/Volume_Scattering/Volume_Scattering_Processes).

**Reply by njroussel** [reply]
I checked the mitsuba 0.5 documentation, `density` should just be `sigma_t` now.
